Moved oauth info to separate file away from jupyter_notebook_config.py

agjwong opened the pull request.

mreid-moz commented:
Please switch to a [common config file format](https://docs.python.org/2/library/configparser.html), and make sure that the Client ID is available from [gist.js](https://github.com/mozilla/jupyter-notebook-gist/blob/master/jupyter-notebook-gist/extensions/gist.js#L54) as well.

mreid-moz commented:
Thanks, it'll be good to get rid of those warnings.

agjwong commented:
Changed config format, thanks for that. Also added some debug messages (success, fail messages to help user). 

Is there some better way to do this?  the gist.js file will have to read the .ini file manually it seems

agjwong commented:
Note: still need to test on Python 2.x due to the ConfigParser module being renamed

mreid-moz commented:
I think there must be an easier way to do this.  According to [this source](http://jupyter-notebook.readthedocs.io/en/latest/frontend_config.html#persisting-configuration-settings), we could persist the oauth config in the `common` section to make it available to the client-rendered js. Can we update that config from the python code, storing **only** the value of `client_id` so we don't expose the client secret to end-users?

Also, once we sort out the preferred method, please update the `README` with updated instructions.

agjwong commented:
This is going to sound quite silly but instead of directly inserting the client id to the jupyter_notebook_config.py file that we did previously, upon loading the extension, it will read in the credentials and store it into the config.  The usage for the .js file remains the same since the client id is in the exact same location as before.  Only now, there are no error messages!

jezdez commented:
I took the liberty to look at the PR after I've worked on #48 and concur with @mreid-moz that using the `common` section is preferable to introducing an own config file. The code to modify that file seems to be the `notebook.services.config.ConfigManager` as used in the current README example of the `~/jupyter_notebook_config.py` already. Just replace the `notebook` with `common`, and we should be good on the setup side. On the frontend side of things I'm not sure how to get the common section though, so either we fetch it by hand from `http://localhost:8888/api/config/common` (or using the official `configmod.ConfigSection.load` JavaScript API, dunno what you'd prefer. Anyways, just my 2 cents :)

agjwong commented:
Thanks for the comment.  I'll have another look tonight when I get back - I do agree that sounds much better, sadly I am having trouble finding decent documentation for the Jupyter config structure even after reading all the manpages/docs provided.

jezdez commented:
@agjwong Yeah, the docs are lacking, just went with reading the code in the notebook repo, too :-/

mreid-moz commented:
I haven't looked into it in detail, but using the ` configmod.ConfigSection.load` API seems preferable to me.

agjwong commented:
After looking at it again, the issue seems to be with these two lines:

```
c.NotebookApp.oauth_client_id = "my_client_id"         # FIXME
c.NotebookApp.oauth_client_secret = "my_client_secret" # FIXME
```

Not so much the ConfigManager code.  Since NotebookApp.oauth_client_id and client_secret are not members of the NotebookApp, you get the error message when running the notebook: 

```
[W 14:32:29.154 NotebookApp] Config option `oauth_client_secret` not recognized by `NotebookApp`.  Did you mean `cookie_secret`?
[W 14:32:29.155 NotebookApp] Config option `oauth_client_id` not recognized by `NotebookApp`.
```
